Add IDL files and empty implementation for Cookie Store API

RupinMittal opened the pull request:
#### 6178f6321703c1007f79e9beeef8d27f8da9eec2
<pre>
Add IDL files and empty implementation for Cookie Store API
<a href="https://bugs.webkit.org/show_bug.cgi?id=258651">https://bugs.webkit.org/show_bug.cgi?id=258651</a>

Reviewed by Chris Dumez.

This adds the window related IDL files and code stubs for the
new CookieStore API.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/cookie-store/CookieInit.h: Added.
* Source/WebCore/Modules/cookie-store/CookieInit.idl: Added.
* Source/WebCore/Modules/cookie-store/CookieListItem.h: Added.
* Source/WebCore/Modules/cookie-store/CookieListItem.idl: Added.
* Source/WebCore/Modules/cookie-store/CookieSameSite.h: Added.
* Source/WebCore/Modules/cookie-store/CookieSameSite.idl: Added.
* Source/WebCore/Modules/cookie-store/CookieStore.cpp: Added.
(WebCore::CookieStore::create):
(WebCore::CookieStore::~CookieStore):
(WebCore::CookieStore::get):
(WebCore::CookieStore::getAll):
(WebCore::CookieStore::set):
(WebCore::CookieStore::remove):
(WebCore::CookieStore::eventTargetInterface const):
(WebCore::CookieStore::scriptExecutionContext const):
* Source/WebCore/Modules/cookie-store/CookieStore.h: Added.
* Source/WebCore/Modules/cookie-store/CookieStore.idl: Added.
* Source/WebCore/Modules/cookie-store/CookieStoreDeleteOptions.h: Added.
* Source/WebCore/Modules/cookie-store/CookieStoreDeleteOptions.idl: Added.
* Source/WebCore/Modules/cookie-store/CookieStoreGetOptions.h: Added.
* Source/WebCore/Modules/cookie-store/CookieStoreGetOptions.idl: Added.
* Source/WebCore/Modules/cookie-store/LocalDOMWindow+CookieStore.idl: Added.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/dom/EventTargetFactory.in:
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::cookieStore):
* Source/WebCore/page/LocalDOMWindow.h:

Canonical link: <a href="https://commits.webkit.org/265669@main">https://commits.webkit.org/265669@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a3b4cc5183036a31a22b545ebc721bd727f37236

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/11580 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11783 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/12144 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/13226 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/11039 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/11601 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/14170 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/11764 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13912 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/11744 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/12602 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/9820 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/13647 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9894 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/10515 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/17655 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/9847 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/10967 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/10669 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/13850 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/10990 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/11065 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/9121 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/11669 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/10241 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/3142 "Passed tests") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/2786 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/14518 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/11999 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/10921 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/2914 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebCore/Modules/cookie-store/CookieInit.h:
```diff
@@ -0,0 +1,44 @@
+    String value;
+    std::optional<DOMHighResTimeStamp> expires;
+    String domain;
+    String path = "/"_s;
```
I think we usually prefer to default initialize like this 
```suggestion
    String path { "/"_s };
```

RupinMittal replied:
Fixed!

rr-codes commented on Source/WebCore/Modules/cookie-store/CookieListItem.h:
```diff
@@ -0,0 +1,45 @@
+    String domain;
+    String path;
+    std::optional<DOMHighResTimeStamp> expires;
+    bool secure;
```
this should have a default value of `false` I believe

cdumez replied:
Then the next data member needs a default value too

RupinMittal replied:
Fixed!

RupinMittal commented on Source/WebCore/CMakeLists.txt:
```diff
@@ -287,6 +287,14 @@ set(WebCore_NON_SVG_IDL_FILES
 
+    Modules/cookie-store/CookieInit.idl
+    Modules/cookie-store/CookieListItem.idl
+    Modules/cookie-store/CookieSameSite.h
```
This is the cause of the build failures. Should be "CookieSameSite.idl". I'll fix this in the next iteration.

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/page/LocalDOMWindow.cpp:
```diff
@@ -2803,4 +2804,9 @@ WebCoreOpaqueRoot root(LocalDOMWindow* window)
 
+CookieStore* LocalDOMWindow::cookieStore()
+{
+    return nullptr;
```
It would be good to lazily construct a CookieStore here, store it as a data member and return a reference to it so that we start to see some tests passing hopefully.

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/LocalDOMWindow+CookieStore.idl:
```diff
@@ -0,0 +1,33 @@
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+ 
+// https://wicg.github.io/cookie-store/
```
Ideally would be https://wicg.github.io/cookie-store/#Window

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStoreGetOptions.idl:
```diff
@@ -0,0 +1,33 @@
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+// https://wicg.github.io/cookie-store/
```
https://wicg.github.io/cookie-store/#dictdef-cookiestoregetoptions

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStoreGetOptions.idl:
```diff
@@ -0,0 +1,33 @@
+// https://wicg.github.io/cookie-store/
+ 
+[
+    EnabledBySetting=CookieStoreAPIEnabled
```
Not needed for dictionaries, they are not JS-exposed types.

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStoreDeleteOptions.idl:
```diff
@@ -0,0 +1,34 @@
+// https://wicg.github.io/cookie-store/
+ 
+[
+    EnabledBySetting=CookieStoreAPIEnabled
```
Not needed for dictionaries.

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStoreDeleteOptions.idl:
```diff
@@ -0,0 +1,34 @@
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+// https://wicg.github.io/cookie-store/
```
https://wicg.github.io/cookie-store/#dictdef-cookiestoredeleteoptions

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStore.idl:
```diff
@@ -0,0 +1,49 @@
+    [ImplementedAs=remove] Promise<undefined> delete(CookieStoreDeleteOptions options);
+
+    [Exposed=Window]
+    attribute EventHandler onchange;
```
I would put this on the previous line:
```
[Exposed=Window] attribute EventHandler onchange;
```

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStore.idl:
```diff
@@ -0,0 +1,49 @@
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+ 
+// https://wicg.github.io/cookie-store/
```
https://wicg.github.io/cookie-store/#CookieStore

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStore.h:
```diff
@@ -0,0 +1,67 @@
+    void remove(const String&, Ref<DeferredPromise>&&);
+    void remove(CookieStoreDeleteOptions&&, Ref<DeferredPromise>&&);
+
+    EventTargetInterface eventTargetInterface() const final;
```
We usually group overrides together with a comment like so:
```
// EventTarget
EventTargetInterface eventTargetInterface() const final;
ScriptExecutionContext* scriptExecutionContext() const final;
void refEventTarget() final { ref(); }
void derefEventTarget() final { deref(); }
```

Note that I marked all of them as `final`. Also, they can be made private.

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStore.h:
```diff
@@ -0,0 +1,67 @@
+
+class DeferredPromise;
+
+class CookieStore : public EventTarget {
```
class can be marked as `final`.

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStore.cpp:
```diff
@@ -0,0 +1,66 @@
+Ref<CookieStore> CookieStore::create() { return adoptRef(*new CookieStore()); }
+CookieStore::~CookieStore() { }
+
+void CookieStore::get(const String&, Ref<DeferredPromise>&&) { }
```
We should reject all promises with `NotSupportedError` to avoid timeouts in the tests.

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStore.cpp:
```diff
@@ -0,0 +1,66 @@
+
+namespace WebCore {
+
+Ref<CookieStore> CookieStore::create() { return adoptRef(*new CookieStore()); }
```
Please do this on multiple lines.

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStore.cpp:
```diff
@@ -0,0 +1,66 @@
+namespace WebCore {
+
+Ref<CookieStore> CookieStore::create() { return adoptRef(*new CookieStore()); }
+CookieStore::~CookieStore() { }
```
`= default;`

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieSameSite.idl:
```diff
@@ -0,0 +1,32 @@
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+// https://wicg.github.io/cookie-store/
```
https://wicg.github.io/cookie-store/#enumdef-cookiesamesite

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStore.cpp:
```diff
@@ -0,0 +1,66 @@
+CookieStore::~CookieStore() { }
+
+void CookieStore::get(const String&, Ref<DeferredPromise>&&) { }
+void CookieStore::get(std::optional<CookieStoreGetOptions>, Ref<DeferredPromise>&&) { }
```
Please do all these on multiple lines please.

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStore.cpp:
```diff
@@ -0,0 +1,66 @@
+void CookieStore::remove(const String&, Ref<DeferredPromise>&&) { }
+void CookieStore::remove(CookieStoreDeleteOptions&&, Ref<DeferredPromise>&&) { }
+
+CookieStore::CookieStore() = default;
```
Could you please move this towards the top of the files? Ideally between the create() and the destructor.

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStore.h:
```diff
@@ -0,0 +1,67 @@
+    void get(Ref<DeferredPromise>&&);
+
+    void getAll(const String&, Ref<DeferredPromise>&&);
+    void getAll(std::optional<CookieStoreGetOptions>, Ref<DeferredPromise>&&);
```
No std::optional, just `CookieStoreGetOptions &&`. You'll get a default-constructed struct from the JS bindings if missing.

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStore.h:
```diff
@@ -0,0 +1,67 @@
+    virtual ~CookieStore();
+
+    void get(const String&, Ref<DeferredPromise>&&);
+    void get(std::optional<CookieStoreGetOptions>, Ref<DeferredPromise>&&);
```
No std::optional

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStore.h:
```diff
@@ -0,0 +1,67 @@
+    static Ref<CookieStore> create();
+    virtual ~CookieStore();
+
+    void get(const String&, Ref<DeferredPromise>&&);
```
Please don't omit the first parameter name, it is not obvious.

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStore.h:
```diff
@@ -0,0 +1,67 @@
+    void get(std::optional<CookieStoreGetOptions>, Ref<DeferredPromise>&&);
+    void get(Ref<DeferredPromise>&&);
+
+    void getAll(const String&, Ref<DeferredPromise>&&);
```
Don't omit first parameter name.

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStore.h:
```diff
@@ -0,0 +1,67 @@
+    void getAll(std::optional<CookieStoreGetOptions>, Ref<DeferredPromise>&&);
+    void getAll(Ref<DeferredPromise>&&);
+
+    void set(const String&, const String&, Ref<DeferredPromise>&&);
```
Don't omit 2 first parameter names.

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieSameSite.h:
```diff
@@ -0,0 +1,36 @@
+
+namespace WebCore {
+
+enum class CookieSameSite : uint8_t {
```
Could be all on one line.

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieListItem.idl:
```diff
@@ -0,0 +1,40 @@
+typedef double DOMHighResTimeStamp;
+ 
+[
+    EnabledBySetting=CookieStoreAPIEnabled
```
Not needed for dictionaries.

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieListItem.idl:
```diff
@@ -0,0 +1,40 @@
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+// https://wicg.github.io/cookie-store/
```
https://wicg.github.io/cookie-store/#dictdef-cookielistitem

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieListItem.h:
```diff
@@ -0,0 +1,45 @@
+
+namespace WebCore {
+
+typedef double DOMHighResTimeStamp;
```
We prefer this in new code:
`using DOMHighResTimeStamp = double;`

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieInit.idl:
```diff
@@ -0,0 +1,39 @@
+typedef double DOMHighResTimeStamp;
+ 
+[
+    EnabledBySetting=CookieStoreAPIEnabled
```
Not needed on dictionaries.

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieInit.idl:
```diff
@@ -0,0 +1,39 @@
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+// https://wicg.github.io/cookie-store/
```
https://wicg.github.io/cookie-store/#dictdef-cookieinit

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStoreDeleteOptions.h:
```diff
@@ -0,0 +1,36 @@
+struct CookieStoreDeleteOptions {
+    String name;
+    String domain;
+    String path = "/"_s;
```
If you apply Richard's comment above for initialization, then please be consistent here. Personally, I think either would be fine as long as we are consistent.

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieInit.h:
```diff
@@ -0,0 +1,44 @@
+    std::optional<DOMHighResTimeStamp> expires;
+    String domain;
+    String path { "/"_s };
+    CookieSameSite sameSite = CookieSameSite::Strict;
```
Please be consistent with initialization. You used curly brackets above and = here.

RupinMittal replied:
Fixed!

cdumez commented on Source/WebCore/Modules/cookie-store/CookieListItem.idl:
```diff
@@ -0,0 +1,39 @@
+dictionary CookieListItem {
+    USVString name;
+    USVString value;
+    // FIXME: Uncomment if you want to expose these attributes
```
Please add a blank line before. Also, comments need to end with a period in WebKit.

I would update the FIXME comment to:
```
// FIXME: These dictionary members are present in the specification. We should evaluate if we want to expose them.
```

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStore.cpp:
```diff
@@ -0,0 +1,93 @@
+    promise->reject(NotSupportedError);
+}
+
+void CookieStore::set(CookieInit&&, Ref<DeferredPromise>&&) { }
```
Should be on multiple lines and reject the promise.

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStore.h:
```diff
@@ -0,0 +1,65 @@
+class CookieStore final : public EventTarget {
+public:
+    static Ref<CookieStore> create();
+    virtual ~CookieStore();
```
nit: we don't need the `virtual` here, it is virtual because the base class's destructor is.

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStore.h:
```diff
@@ -0,0 +1,65 @@
+
+#pragma once
+
+#include "CookieInit.h"
```
We should forward declare those first 3.

cdumez commented on Source/WebCore/page/LocalDOMWindow.cpp:
```diff
@@ -2803,4 +2804,10 @@ WebCoreOpaqueRoot root(LocalDOMWindow* window)
 
+CookieStore& LocalDOMWindow::cookieStore()
+{
+    m_cookieStore = CookieStore::create();
```
You'll keep re-creating the cookie store, that's not good.

Should be:
```
if (!m_cookieStore)
    m_cookieStore = CookieStore::create();
return *m_cookieStore;
```

rr-codes commented on Source/WebCore/Modules/cookie-store/CookieListItem.h:
```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "CookieSameSite.h"
+#include <optional>
```
this include can be removed

rr-codes commented on Source/WebCore/Modules/cookie-store/CookieInit.h:
```diff
@@ -0,0 +1,44 @@
+struct CookieInit {
+    String name;
+    String value;
+    std::optional<DOMHighResTimeStamp> expires;
```
does this actually need to be optional?

whsieh replied:
This is `optional` because the `expires` attribute is optional in the IDL.

whsieh commented on Source/WebCore/Modules/cookie-store/CookieListItem.h:
```diff
@@ -0,0 +1,40 @@
+
+namespace WebCore {
+
+using DOMHighResTimeStamp = double;
```
Perhaps instead of redefining this, we could include `DOMHighResTimeStamp.h`?

cdumez replied:
Yes

cdumez commented on Source/WebCore/Modules/cookie-store/CookieListItem.h:
```diff
@@ -0,0 +1,38 @@
+
+#pragma once
+
+#include "CookieSameSite.h"
```
This include is not needed.

cdumez commented on Source/WebCore/Modules/cookie-store/CookieListItem.h:
```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include "CookieSameSite.h"
+#include "DOMHighResTimeStamp.h"
```
ditto.

cdumez reviewed (approved):
LGTM with test rebaselined.

RupinMittal committed:
Add IDL files and empty implementation for Cookie Store API
https://bugs.webkit.org/show_bug.cgi?id=258651

Reviewed by Chris Dumez.

This adds the window related IDL files and code stubs for the
new CookieStore API.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/cookie-store/CookieInit.h: Added.
* Source/WebCore/Modules/cookie-store/CookieInit.idl: Added.
* Source/WebCore/Modules/cookie-store/CookieListItem.h: Added.
* Source/WebCore/Modules/cookie-store/CookieListItem.idl: Added.
* Source/WebCore/Modules/cookie-store/CookieSameSite.h: Added.
* Source/WebCore/Modules/cookie-store/CookieSameSite.idl: Added.
* Source/WebCore/Modules/cookie-store/CookieStore.cpp: Added.
(WebCore::CookieStore::create):
(WebCore::CookieStore::~CookieStore):
(WebCore::CookieStore::get):
(WebCore::CookieStore::getAll):
(WebCore::CookieStore::set):
(WebCore::CookieStore::remove):
(WebCore::CookieStore::eventTargetInterface const):
(WebCore::CookieStore::scriptExecutionContext const):
* Source/WebCore/Modules/cookie-store/CookieStore.h: Added.
* Source/WebCore/Modules/cookie-store/CookieStore.idl: Added.
* Source/WebCore/Modules/cookie-store/CookieStoreDeleteOptions.h: Added.
* Source/WebCore/Modules/cookie-store/CookieStoreDeleteOptions.idl: Added.
* Source/WebCore/Modules/cookie-store/CookieStoreGetOptions.h: Added.
* Source/WebCore/Modules/cookie-store/CookieStoreGetOptions.idl: Added.
* Source/WebCore/Modules/cookie-store/LocalDOMWindow+CookieStore.idl: Added.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/dom/EventTargetFactory.in:
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::cookieStore):
* Source/WebCore/page/LocalDOMWindow.h:

Canonical link: https://commits.webkit.org/265669@main
